libs/contrib/Text/Lexer.idr

```diff
@@ -88,13 +88,30 @@ choice : (xs : List Lexer) -> {auto ok : NonEmpty xs} -> Lexer
 choice (x :: [])          = x
 choice (x :: xs@(_ :: _)) = x <|> choice xs
 
+||| Repeat the sub-lexer `l` zero or more times until the lexer
+||| `stopBefore` is encountered. `stopBefore` will not be consumed.
+||| Not guaranteed to consume input.
+export
+manyUntil : (stopBefore : Recognise c) -> (l : Lexer) -> Recognise False
+manyUntil stopBefore l = many (reject stopBefore <+> l)
+
+||| Repeat the sub-lexer `l` zero or more times until the lexer
+||| `stopAfter` is encountered, and consume it. Guaranteed to
+||| consume if `stopAfter` consumes.
+export
+manyThen : (stopAfter : Recognise c) -> (l : Lexer) -> Recognise c
+manyThen stopAfter l = manyUntil stopAfter l <+> stopAfter
+
 ||| Recognise many instances of `l` until an instance of `end` is
 ||| encountered.
 |||
 ||| Useful for defining comments.
 export
 manyTill : (l : Lexer) -> (end : Lexer) -> Recognise False
 manyTill l end = end <|> opt (l <+> manyTill l end)
+%deprecate manyTill "Prefer `manyUntil` and `manyThen`. Argument order is flipped. \
```

Comment:
Idris supports multiline strings using the following syntax: `""" """`. It would be preferable if such syntax is used.

Reply:
`%deprecate` doesn't seem to work with triple quotes. Using them was my initial thought, too, but I couldn't the compiler it to accept it.

Reply:
Urm I thought it did... This must be a parser restriction...I am tempted to say reduce the statement so that it just points to the alternatives, or fix the parser....

Reply:
I'll trim down this message once I've added `lineComment` (coming soon).